src/__tests__/rust-plugin-test.js

```diff
@@ -15,12 +56,22 @@ if (rustInstalled) {
     }, 10000);
 
     it('calc rust example should build, also output must match expected value', () => {
-      const runResult = shelljs.exec('cargo run --quiet', {
+      let runResult = shelljs.exec('cargo run --quiet', {
         silent: true,
         cwd: rustCalcDir,
       });
 
-      expect(runResult.stderr).toEqual('');
+      if (runResult.code !== 0) {
+        // something went wrong, rerun command with full debug output
+        runResult = shelljs.exec('cargo run', {
+          silent: false,
+          cwd: rustCalcDir,
+        });
+        console.error(runResult.stderr);
```

Comment:
Since this requires Rust to be installed in the first place on user machine, should we run this test only when it's available?

Since Syntax is language-agnostic, it should not require Rust or any other runtimes to be installed to generate parsers or run its tests. OTOH, for those who have Rust installed, the test would be good to run.

Reply:
> should we run this test only when it's available?

It acts like it right now. Look at line 50: `if (makeInstalled && rustInstalled && minimumVersionSatisfied) {`.

If rust is not installed or other requirements are not satisfied test will always pass with warning (on line 86)

Reply:
Checks for rust installation were in code before this PR as well. But these checks were inaccurate, for example check for rust minimum version was not done.

Reply:
So, code you point here will not run at all if any of conditions are true:

1. rust is not installed
2. make is not installed
3. rust has version less than 1.31

Reply:
I see, make sense, thanks!